### PR TITLE
Add possibility to expose Function in Functions view

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Add possibility to expose Function in Functions view:
  - use Luigi to expose API Rules `create` and `edit` views in Function details view
- Update Service Details view with new changes in ApiRules view

**Related issue(s)**
https://github.com/kyma-project/console/issues/1827

### CODE REVIEW

**Comment by Wawrzyn321** [comment]
Updating APIRule fails with 422 - please include `generation` in mutation parameters (it will be supplied with APIRule query).

**Comment by Wawrzyn321** [comment]
I've noticed `redirectPath` and using `split`+`join` to get fix '/'. Wouldn't be better to use `encodeURI` before passing it to Luigi params and then `decodeURI`?

File: core-ui/src/components/ApiRules/gql/test/useApiRulesQuery.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    );
+
+    const message = formatMessage(GQL_QUERIES.API_RULES.ERROR_MESSAGE, {
+      lambdaName: lambdaMock.name,
+      error: `Network error: ${TESTING_STATE.ERROR}`,
+    });
+
+    await wait(() => {
+      expect(getByText(message)).toBeInTheDocument();
+    });
```
Comment (Wawrzyn321) [inline core-ui/src/components/ApiRules/gql/test/useApiRulesQuery.test.js:63]: Wouldn't `findByText` make more sense here?

File: core-ui/src/components/ApiRules/ApiRuleDetails/ApiRuleDetails.js [hunk 2]
```diff
@@ -93,34 +91,34 @@ function navigateToEditView(apiRuleName) {
...
+const breadcrumbItems = [{ name: 'API Rules', path: '/' }, { name: '' }];
 
-  const navigateToService = () =>
-    LuigiClient.linkManager()
-      .fromContext('namespaces')
-      .navigate(`services/details/${data.service.name}`);
+function ApiRuleDetailsHeader({ apiRule }) {
+  const name = apiRule.name;
+  const { openedInModal = false } = LuigiClient.getNodeParams() || {};
+  const openedInModalBool = openedInModal.toString().toLowerCase() === 'true';
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/ApiRules/ApiRuleDetails/ApiRuleDetails.js:99]: Luigi send context by postMessage and it cast everything to string, also objects and boolean...

File: core-ui/src/components/ApiRules/ApiRuleForm/ApiRuleFormHeader/ApiRuleFormHeader.js [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
 
+import './ApiRuleFormHeader.scss';
+
 const ApiRuleFormHeader = ({
   handleSave,
   isValid,
   title,
   saveButtonText,
   breadcrumbItems,
+  serviceName = undefined,
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply core-ui/src/components/ApiRules/ApiRuleForm/ApiRuleFormHeader/ApiRuleFormHeader.js:16]: I changed to empty string, but in this case is unnecessary. Please see that serviceName is only used when it isn't empty or undefined.

File: core-ui/src/components/Lambdas/constants.js [hunk 4]
```diff
@@ -148,6 +148,9 @@ export const GQL_QUERIES = {
   CONFIG_MAP: {
     ERROR_MESSAGE: `Couldn't fetch "{configMapName}" ConfigMap due to this error: {error}`,
   },
+  SERVICE: {
+    ERROR_MESSAGE: `Couldn't fetch "{serviceName}" Service due to this error: {error}`,
```
Comment (kazydek) [inline core-ui/src/components/Lambdas/constants.js:152]: As I see [here](https://github.com/kyma-project/console/blob/a6a57fe23fc9370b12f9f6d9c94ef7319f13843a/core-ui/src/components/Lambdas/constants.js#L131), when there was one particular resource we just added its name. I talked to other TWs to have a consistent approach here and we all agreed to use this pattern for all such cases when you have a resource and its name in the **singular form**:"

`Couldn't fetch ConfigMap "{configMapName}" due to this error: {error}`
`Couldn't fetch API Rule "{serviceName}" due to this error: {error}`
`Couldn't fetch Function "{lambdaName}" due to this error: {error}`

...for all cases in these examples: configMapName, serviceName, and [lambdaName](https://github.com/kyma-project/console/blob/a6a57fe23fc9370b12f9f6d9c94ef7319f13843a/core-ui/src/components/Lambdas/constants.js#L131).

File: core-ui/src/components/ApiRules/constants.js [hunk 5]
```diff
@@ -1 +1,59 @@
...
+    TEXT: 'Expose {type}',
+  },
+  CREATE_MODAL: {
+    TITLE: 'Create API Rule',
+    OPEN_BUTTON: {
+      TEXT: 'Expose API',
+    },
+  },
+  EDIT_MODAL: {
+    TITLE: 'Edit "{apiRuleName}" API Rule',
```
Comment (kazydek) [inline core-ui/src/components/ApiRules/constants.js:30]: `Edit API Rule "{apiRuleName}"`

File: core-ui/src/components/ApiRules/constants.js [hunk 6]
```diff
@@ -1 +1,59 @@
...
+  },
+};
+
+export const GQL_MUTATIONS = {
+  DELETE_API_RULE: {
+    SUCCESS_MESSAGE: `API Rule was successfully deleted`,
+    ERROR_MESSAGE: `Couldn't delete the API Rule due to this error: {error}`,
+    CONFIRM_MODAL: {
+      TITLE: `Delete API Rule`,
+      MESSAGE: `Are you sure you want to delete the "{apiRuleName}" API Rule?`,
```
Comment (kazydek) [inline core-ui/src/components/ApiRules/constants.js:56]: `Are you sure you want to delete API Rule "{apiRuleName}"?`

**Pull request merged**